core/src/main/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package com.opensymphony.xwork2.interceptor;
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.ActionInvocation;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CSRFPreventionInterceptor extends AbstractInterceptor {
+
+    private List<String> exemptedPaths = new ArrayList<String>();
+    private DefaultResourceIsolationPolicy fmPolicy = new DefaultResourceIsolationPolicy();
```

Comment:
These two fields can be final. Struts interceptors **must** be [thread safe](https://struts.apache.org/core-developers/writing-interceptors.html)!
```suggestion
    private final List<String> exemptedPaths = new ArrayList<String>();
    private final DefaultResourceIsolationPolicy fmPolicy = new DefaultResourceIsolationPolicy();
```